### DISCUSSION TITLE
Use ChromaDB in server mode

### DISCUSSION THREAD

**Discussion by trguduru** [vanna-ai/vanna#499] [Q&A] [answered]
Hello,

I'm attempting to utilize ChromaDB in server mode and employ it in ChromaDB_VectorStore instead of using an in-memory/persistent filesystem within the Flask app.

I attempted to pass the HTTP address of the running Chroma server as the path in the config, but it didn't work...

Is there an option for this? I couldn't find this option; the only ChromaDB client options I see are persistent or in-memory.

**Comment by zainhoda** [comment, marked as answer]
You can actually pass in the chroma client into that parameter to set the client directly:

https://github.com/vanna-ai/vanna/blob/main/src/vanna/chromadb/chromadb_vector.py#L37

**Reply by trguduru** [reply]
Awesome, earlier i tried with passing the chromaDB http address but it sounds like its expecting an instance of Chroma Client as a config ... 

Tried with the following and its working fine ..

```
chroma_client = chromadb.HttpClient(host='localhost', port=8000)
vn = MyVanna(config={'model': 'mistral', "client":chroma_client})
```
I was expecting to provide an HTTP address and build the client within Vanna implementation.

**Reply by geraldineLTC** [reply]
@zainhoda @trguduru 
Below is my code, but failed to get through it, got HTTP status error 503 "ValueError: Could not connect to tenant default_tenant"

class MyVanna(ChromaDB_VectorStore, GoogleGeminiChat):
   def __init__(self, config=None):
       ChromaDB_VectorStore.__init__(self, config=config)
       GoogleGeminiChat.__init__(self, config={'api_key': 'my api_key', 'model': 'gemini-1.5-flash-latest'})
chroma_client = HttpClient(host='127.0.0.1', port=8000, settings=Settings(allow_reset=True, anonymized_telemetry=False))
vn = MyVanna(config={"client":chroma_client,'path': 'C:/Containers/Chromadb/'})

and i am sure my local chromadb is on, can access through browser. any idea?
